packages/structures/src/channels/GroupDMChannel.ts

```diff
@@ -0,0 +1,47 @@
+import type { APIGroupDMChannel, ChannelType } from 'discord-api-types/v10';
+import { Mixin } from '../Mixin.js';
+import type { MixinTypes } from '../MixinTypes.d.ts';
+import { kData } from '../utils/symbols.js';
+import { Channel } from './Channel.js';
+import { ChannelOwnerMixin } from './mixins/ChannelOwnerMixin.js';
+import { DMChannelMixin } from './mixins/DMChannelMixin.js';
+import { TextChannelMixin } from './mixins/TextChannelMixin.js';
+
+export interface GroupDMChannel<Omitted extends keyof APIGroupDMChannel | '' = ''>
+	extends MixinTypes<
+		Channel<ChannelType.GroupDM>,
+		[DMChannelMixin<ChannelType.GroupDM>, TextChannelMixin<ChannelType.GroupDM>, ChannelOwnerMixin<ChannelType.GroupDM>]
+	> {}
+
+export class GroupDMChannel<Omitted extends keyof APIGroupDMChannel | '' = ''> extends Channel<
+	ChannelType.GroupDM,
+	Omitted
+> {
+	public constructor(data: Omit<APIGroupDMChannel, Omitted>) {
+		super(data);
+		this.optimizeData(data);
+	}
+
+	/**
+	 * The icon hash of the group DM.
+	 */
+	public get icon() {
```

Comment:
Since these classes aren't being used in implementing libraries maybe we should pull these out into a "mixin" even though it's for a single channel type?